altair/utils/schemapi.py

```diff
@@ -1087,10 +1087,7 @@ def _default_wrapper_classes(cls) -> Iterator[type[SchemaBase]]:
 
     @classmethod
     def from_dict(
-        cls: type[TSchemaBase],
-        dct: dict[str, Any],
-        validate: bool = True,
-        _wrapper_classes: Iterable[type[SchemaBase]] | None = None,
+        cls: type[TSchemaBase], dct: dict[str, Any], validate: bool = True, **kwds: Any
```

Comment:
See [comment](https://github.com/vega/altair/pull/3480/files#r1679062039)

Reply:
I'm not using pyright so not sure what error was raised here but could we just ignore it? At least looking at the code, we lose the information of what type `_wrapper_classes` can be and there is no need to be able to pass other keyword arguments. I'd prioritize readability over helping out pyright but maybe I'm missing something?

Reply:
> I'm not using pyright so not sure what error was raised here but could we just ignore it? At least looking at the code, we lose the information of what type `_wrapper_classes` can be and there is no need to be able to pass other keyword arguments. I'd prioritize readability over helping out pyright but maybe I'm missing something?

So the original error was a `mypy` one in `Chart.from_dict`. 
I've confused myself trying to follow the links I created (sorry), for reference this is what I was fixing:



https://github.com/vega/altair/blob/9afd374a83f954e79a5fd6e4f8a8a9c4e5381c04/altair/vegalite/v5/api.py#L2885-L2890

The problem is defined in the mypy docs [incompatible-overrides](https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides)

Dropping `_wrapper_classes` made sense to me as a solution:
- I cannot find any `altair` code using the argument
- The only modification to the default occurs in `VegaLiteSchema` https://github.com/vega/altair/blob/9afd374a83f954e79a5fd6e4f8a8a9c4e5381c04/altair/vegalite/v5/schema/core.py#L454
	- this redeclares `_default_wrapper_classes` with `Iterator[type[Any]]`
- My assumption that the reason for the ignore, was to avoid displaying `_wrapper_classes` as a parameter for `Chart.from_dict`
- This makes the signatures of all derived classes more flexible
	- which I think is more important than, *what appears to me*, a niche feature

I'm doubting myself now, but does that additional context help explain my choice better?

Reply:
Interesting. Given that it's not used anywhere in the `altair` code base and it's prefixed with a `_`, what do you think about removing this completely and not accepting any arbitrary `kwds` at all? I'd be very surprised if anyone else has used this.

Reply:
> Interesting. Given that it's not used anywhere in the `altair` code base and it's prefixed with a `_`, what do you think about removing this completely and not accepting any arbitrary `kwds` at all? I'd be very surprised if anyone else has used this.

I wouldn't be opposed. 

I'll have a look to see if maybe this was used somewhere historically first. 
I find it quite strange that `_wrapper_classes` appears in the docstring - as well as appearing to be private.

Reply:
It was added in `v2.0.0` here, at the same time as the `_default_wrapper_classes` method

- https://github.com/vega/altair/commit/803f42e4d322f1ddcb6dc44621ecdb56a742a8d1

Reply:
Based on this PR description, I think we should remove it @binste

> This is a maintenance fix for a subtle bug I came across. The issue was that when running `from_dict`, altair would consider any subclass of `SchemaBase` as a potential wrapper for the generated class hierarchy.
> 
> What this meant is that if you, say, imported both **vega-lite **and **vega** wrappers, you could get cross-talk between them when using from_dict if they had common schemas.
> 
> This PR changes the auto-generated code so that each set of wrappers inherits from its own special base class, which ensures that this kind of cross-talk does not happen.

There are only `vega-lite` wrappers in `v5.0.0`, so this is no longer a concern

Reply:
Nice git-archeology! ;) Thanks for digging it out and I agree, as we've removed the Vega-related code parts in v5, we can get rid of this.